core/divesoft.cpp

```diff
@@ -13,27 +13,15 @@
 #include "format.h"
 #include "libdivecomputer.h"
 
-// As supplied by Divesoft
-static constexpr std::string_view divesoft_liberty_serial_prefix = "7026";
-static constexpr std::string_view divesoft_freedom_serial_prefix = "7044";
-static constexpr std::string_view divesoft_freedom_plus_serial_prefix = "7273";
-
 // From libdivecomputer
 static constexpr int divesoft_liberty_model = 10;
 static constexpr int divesoft_freedom_model = 19;
 
 int divesoft_import(const std::string &buffer, struct divelog *log)
 {
-	std::string model_identifier = buffer.substr(52, 4);
-	int model = 0;
-	if (model_identifier == divesoft_liberty_serial_prefix)
-		model = divesoft_liberty_model;
-	else if (model_identifier == divesoft_freedom_serial_prefix || model_identifier == divesoft_freedom_plus_serial_prefix)
-		model = divesoft_freedom_model;
-
 	device_data_t devdata;
 	devdata.log = log;
-	int ret = prepare_device_descriptor(model, DC_FAMILY_DIVESOFT_FREEDOM, devdata);
+	int ret = prepare_device_descriptor(divesoft_freedom_model, DC_FAMILY_DIVESOFT_FREEDOM, devdata);
```

Comment:
This is incorrect for all devices except for the Divesoft Freedom. Please fix.

Reply:
> I can't do that without copying significant parts of the parser code.

Couldn't the parser just report. back the correct dive computer with `DC_EVENT_DEVINFO`?
Alternatively, the current implementation seems to be what is needed to make this work correctly - it is just incomplete. And it could probably have a fall through case if the serial number does not match a known pattern.

> Even if we do that, I don't have a complete list of the product numbers and their corresponding model number.

We could just ask Divesoft for a list - when I originally implemented this that worked like a charm. It just looks like they have added more different models in the meantime.